react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -159,35 +164,36 @@ const traverseExpression = (
   return expressionResults;
 };
 
-export const generateRuleBasedOutputs = (
+export const generateRuleBasedOutputs = async (
   model: CogniteCadModel,
   contextualizedAssetNodes: Asset[],
   assetMappings: AssetMapping3D[],
   ruleSet: RuleOutputSet | Record<string, any> | FdmPropertyType<Record<string, any>>
-): void => {
+): Promise<AssetStylingGroupAndStyleIndex[]> => {
   const outputType = 'color'; // for now it only supports colors as the output
 
   const ruleWithOutputs = ruleSet?.rulesWithOutputs;
+  return await Promise.all(
+    ruleWithOutputs?.map(async (ruleWithOutput: { rule: Rule; outputs: RuleOutput[] }) => {
+      const { rule, outputs } = ruleWithOutput;
+      // Starting Expression
+      const expression = rule.expression;
 
-  ruleWithOutputs?.forEach(async (ruleWithOutput: { rule: Rule; outputs: RuleOutput[] }) => {
-    const { rule, outputs } = ruleWithOutput;
-    // Starting Expression
-    const expression = rule.expression;
-
-    const outputSelected = outputs.find(
-      (output: { type: string }) => output.type === outputType
-    ) as ColorRuleOutput;
+      const outputSelected = outputs.find(
+        (output: { type: string }) => output.type === outputType
+      ) as ColorRuleOutput;
```

Comment:
I wonder if we can avoid this cast :thinking: Would the type system be able to deduce the type itself if we wrote `'color'` instead of `outputType` here? If not, not critical

Reply:
Yeah, thanks for reminding me. 

Changed here:
https://github.com/cognitedata/reveal/pull/4296/commits/68ddb1037fa2a1bfe9fd9ea6ad07e09522d2865c